folsom/src/main/java/com/spotify/folsom/AbstractRawMemcacheClient.java

```diff
@@ -15,23 +15,31 @@
  */
 package com.spotify.folsom;
 
-import com.google.common.eventbus.EventBus;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 
 public abstract class AbstractRawMemcacheClient implements RawMemcacheClient {
-  private final EventBus eventBus = new EventBus("folsom eventbus");
+  // Value type does not matter here, since we're using this as a concurrent set.
+  private final Map<ConnectionChangeListener, String> listeners = new ConcurrentHashMap<>();
```

Comment:
Could use `ConcurrentHashMap.newKeySet()` to make it an actual Set.

Reply:
Nice, I was not aware of that. Will try that out